zbuf/combiner.go

```diff
@@ -55,9 +56,12 @@ type combiner struct {
 	hol     []*zng.Record
 	done    []bool
 	less    RecordCmpFn
+	mu      sync.Mutex
```

Comment:
Nits: Put the mutex immediately before done, and either add a `// protects done` comment on the same line or change the name to doneMu.